### PR TITLE
Custom table component for audit logs - WIP

### PR DESCRIPTION
**Pull request by NareshGopal**
Signed-off-by: Naresh Gopalakrishnan <naresh.gopalakrishnan@seagate.com>

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

**Comment by NareshGopal** [comment]
What can be expected in this commit?
1. Table will be displayed properly 
2. Default pagination is used and it works
3. Search and filter components are displayed on top but is won't function

File: gui/src/components/widgets/cortx-table.vue [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+                  'tableheader',
+                  header.sortable ? 'cortx-cursor-pointer' : ''
+                ]"
+              >
+                <span>{{ header.text }}</span>
+              </th>
+            </tr>
+          </template>
+
+          <!-- <template v-slot:item.timestamp="props">
```
Comment (kupranay) [inline gui/src/components/widgets/cortx-table.vue:64]: remove commented code

File: gui/src/components/widgets/cortx-table.vue [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+                <td
+                v-for="[key, value] in Object.entries(item)"
+                :key="item.request_id"
+                >
+                {{ value }}
+                </td>
+            </tr>
+          </template>
+
+          <!-- <template #footer="{props}">
```
Comment (kupranay) [inline gui/src/components/widgets/cortx-table.vue:79]: remove commented code

**Review by kupranay — CHANGES_REQUESTED** [review]

File: gui/src/components/maintenance/auditlog.vue [hunk 3]
```diff
@@ -255,8 +290,55 @@ export default class CortxAuditLog extends Vue {
       text: "Response Code",
```
Comment (kupranay) [inline gui/src/components/maintenance/auditlog.vue:290]: { 
{
{
text_id:"timestamp"
text: "Timestamp",
      value: "timestamp",
      sortable: true,
      type: "time",
     show:"true/false",
    
    },

in your auditlog component code
check id is present in en.json if yes then display that value if not fallback to "text" field

**Review by kupranay — CHANGES_REQUESTED** [review]

File: gui/src/components/maintenance/cortx-auditlog.vue [hunk 4]
```diff
@@ -293,11 +197,13 @@ export default class CortxAuditLog extends Vue {
   }
 
   public async getAuditLogs() {
+    debugger;
```
Comment (kupranay) [inline gui/src/components/maintenance/cortx-auditlog.vue:200]: remove debugger

File: gui/src/components/widgets/cortx-table.vue [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+  get listeners() {
+    return {
+      ...this.$listeners,
+    };
+  }
+}
+
+</script>
+
+<style>
```
Comment (kupranay) [inline gui/src/components/widgets/cortx-table.vue:131]: export it as a separate css file

**Review by kupranay — CHANGES_REQUESTED** [review]

File: gui/src/components/widgets/cortx-table.vue [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+                  >
+                    mdi-filter
+                  </v-icon>
+                  <v-select
+                   multiple 
+                   outlined
+                   hide-details 
+                   dense 
+                   color="csmprimary"
+                   :items="['Remote IP', 'Path','Method', 'User Agent']"
```
Comment (kupranay) [inline gui/src/components/widgets/cortx-table.vue:40]: items should be coming from calling component

File: gui/src/components/maintenance/cortx-auditlog.vue [hunk 7]
```diff
@@ -70,79 +70,20 @@
...
         <span class="cortx-text-bold cortx-text-lg" id="s3auditlogtext">{{ $t("maintenance.logs") }}</span>
         <v-divider class="my-2"></v-divider>
         <template v-if="auditLog.logs.length > 0">
-          <span
-            class="mb-1 d-block"
-            v-for="(log, index) in auditLog.logs"
-            :key="index"
-            id="auditlog-data"
-          >{{ log }}</span>
+          <CortxTable
```
Comment (kupranay) [inline gui/src/components/maintenance/cortx-auditlog.vue:78]: name of CortxTable should be changed to cortx-v-data-dablt

**Review by kupranay — APPROVED** [review]

**Pull request closed without merging**